server/src/main/java/connection/Server.java

```diff
@@ -42,12 +44,15 @@ public void start() {
                 Socket clientSocket = serverSocket.accept();
                 LOGGER.info(clientSocket.toString() + " connected");
                 MessagesFromClientHandler messagesFromClientHandler = new MessagesFromClientHandler(clientSocket, this);
+                fromClientHandlers.add(messagesFromClientHandler);
+                LOGGER.info(fromClientHandlers);
```

Comment:
createServer() method is growing continously. You can think about extracting some parts of code into separate methods. It should be easier to read it later too.